modules/library/src/main/scala/zio/elasticsearch/executor/response/SearchWithAggregationsResponse.scala

```diff
@@ -92,6 +92,8 @@ private[elasticsearch] final case class SearchWithAggregationsResponse(
                       TermsAggregationResponse.decoder.decodeJson(data.toString).map(field.split("#")(1) -> _)
                     case str if str.contains("value_count#") =>
                       ValueCountAggregationResponse.decoder.decodeJson(data.toString).map(field.split("#")(1) -> _)
+                    case str if str.contains("weighted_avg#") =>
+                      WeightedAvgAggregationResponse.decoder.decodeJson(data.toString).map(field.split("#")(1) -> _)
```

Comment:
Put this first and remove `&& !str.contains("weighted_avg#")` from the first case.